saleor/discount/utils.py

```diff
@@ -46,32 +46,26 @@ def remove_voucher_usage_by_customer(voucher, customer_email):
         voucher_customer.delete()
 
 
-def are_product_collections_on_sale(product, discount: DiscountInfo):
-    """Check if any collection is on sale."""
-    discounted_collections = discount.collection_ids
-    product_collections = set(c.id for c in product.collections.all())
-    return product_collections.intersection(discounted_collections)
-
-
-def get_product_discount_on_sale(product, discount: DiscountInfo):
+def get_product_discount_on_sale(product, product_collections, discount:DiscountInfo):
```

Comment:
You need to update `test_sale_applies_to_correct_products` as well 👍

Reply:
updated